documentation/TODO.md

```diff
@@ -0,0 +1,166 @@
+
+### CDI (Sebastian)
+* Move under an overarching Integrations chapter
+* Make sure we mention split artifacts (embedded and remote)
+* Prevent mentioning configuration overriding 2 times (one for embedded and one for remote)
+
+### Spring (Sebastian)
+* Move under an overarching Integrations chapter
+* Move to its own paragraph under Integrations
+* Do not mention EHCache… it’s not relevant
+* Mention proper artifacts (spring4)
+
+### Remote Protocols (Tristan)
+* The chapter is currently called “Server modules” but this is confusing as we are really discussing the remoting protocols and neither the WildFly modules nor Infinispan Server
+* This should be turned upside down, looking at the protocol side first, and then detailing the server-side implementation aspects.
+
+### Management Tooling (Vittorio)
+* Remove the RHQ section and other mentions of it
+* Move the CLI chapter under here
+
+### Custom Interceptors (Dan)
+* Since we are describing an advanced API this needs to be moved further up
+* Document the new async design and provide some useful examples
+
+### Running on AWS (Sebastian + Gustavo)
+* Actually refactor this chapter into a more general chapter about running in different cloud/container environments
+* AWS
+* Docker
+* OpenShift / Kubernetes
+* Google Compute Engine
+* Azure
+* ...
+
+### WildFly Modules (Gustavo)
+* Move under an overarching Integrations chapter
+
+### Default values for properties (Ryan)
+* Remove this chapter and find a way to generate the information into the configuration reference
+
+### HTTP session clustering and caching (Sebastian)
+* Move under an overarching Integrations chapter
+* Remove the  open tasks
+* Mention Spring Session
+* Eventually will also contain Paul’s Tomcat adapter
+
+### Hadoop + Spark
+* Move under an overarching Integrations chapter
+* ...
+
+### JCache
+* Move under an overarching Integrations chapter
+* Remote JCache configuration, example of properties (at least IP:port)
+
+### Second level cache guide (Galder)
+* Place under Integrations
+
+
+## Server Guide (Tristan)
+
+### Console (Vladimir)
+* The console should get a dedicated chapter outlining the different parts (Clusters, Nodes, Caches, Configuration)
+* Don’t go overboard with screenshots
+* Reference the video
+
+## JavaDoc
+* Ensure that all public APIs have JavaDoc
+* Ensure that all parameters are documented with the @param doclet
+* Can this be somewhat automated via checkstyle  ?
+
+## FAQs
+* We get many recurring questions on the forum / stackoverflow / IRC / etc. Try to distill some of the most significant ones you have answered successfully.
+* Even questions that are related to product (sme-jdg, customer portal, etc) can be of use (obviously purged from any product reference)
+* Things we CANNOT do are also important, and we should point out any limitations
+
+## Performance Guide (Sebastian)
+* This should be linked in the appropriate places from the User Guide
```

Comment:
There's still a lot of ground to cover for query performance guide, I'll add it next week